drivers/spi/spi_gd32.c

```diff
@@ -272,25 +338,38 @@ int spi_gd32_init(const struct device *dev)
 		return ret;
 	}
 
+#if IS_ENABLED(CONFIG_SPI_GD32_INTERRUPT)
+	cfg->irq_configure(dev);
+#endif
+
 	spi_context_unlock_unconditionally(&data->ctx);
 
 	return 0;
 }
 
-#define GD32_SPI_INIT(idx)							\
-	PINCTRL_DT_INST_DEFINE(idx);						\
-	static struct spi_gd32_data spi_gd32_data_##idx = {			\
-		SPI_CONTEXT_INIT_LOCK(spi_gd32_data_##idx, ctx),		\
-		SPI_CONTEXT_INIT_SYNC(spi_gd32_data_##idx, ctx),		\
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(idx), ctx)		\
-	};									\
-	static struct spi_gd32_config spi_gd32_config_##idx = {			\
-		.reg = DT_INST_REG_ADDR(idx),					\
-		.rcu_periph_clock = DT_INST_PROP(idx, rcu_periph_clock),	\
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),			\
-	};									\
-	DEVICE_DT_INST_DEFINE(idx, &spi_gd32_init, NULL, &spi_gd32_data_##idx,	\
-			      &spi_gd32_config_##idx, POST_KERNEL,		\
-			      CONFIG_SPI_INIT_PRIORITY, &spi_gd32_driver_api);
+#define GD32_IRQ_CONFIGURE(idx)                                                                    \
+	static void spi_gd32_irq_configure_##idx(void)                                             \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority), spi_gd32_isr,           \
+			    DEVICE_DT_INST_GET(idx), 0);                                           \
```

Comment:
Long lines, you can put one parameter per line to achieve 80 cols

Reply:
Fixed it.